app/src/main/java/bitcoin/wallet/BaseActivity.kt

```diff
@@ -24,13 +25,14 @@ abstract class BaseActivity : AppCompatActivity() {
 
     override fun onResume() {
         super.onResume()
+        if (this !is PinActivity) {
```

Comment:
It is better to have a flag in `BaseActivity` like `requirePinUnlock`. `PinModule` will start depending also on this flag. This property will be `true` by default. PinActivity will override it to `false`. In this case `BaseActivity` will not know about the `PinActivity`.

Reply:
good point